medcat/config.py

```diff
@@ -15,13 +22,73 @@ def workers(workers_override: Optional[int] = None) -> int:
     return max(cpu_count() - 1, 1) if workers_override is None else workers_override
 
 
-class ConfigMixin(object):
+class FakeDict(object):
```

Comment:
I guess inheriting from object is a habit from python2 where this made _some_ difference. It's definitely not needed here.

As for inheriting from a `dict` (or `UserDict`), it really depends on whether or if we want to keep using it as such.
Right now I only implemented the methods that I could see being used in the project for dict based approach.
However, if we inherit from dict, we would need to overwrite a bunch more of the methods to make sure the beahviour would be predictable.
Plus, some parts of a regular dict would not make sense in the current context. For instance, using `pop` to remove one of the fields.

But then again, if we don't want to have the option of using the `MixingConfig` instances as a `dict`, we would be better off forcing that behaviour by not providing the methods overwritten by `FakeDict`. I've mostly kept that in for backwards compatibility and because I don't know where else parts of the config may be used as a dicts.
So I don't think the current behaviour is ideal, either.

So as I see it, we have a number of options going forward:
- Keep full `dict` functionality
  - Inherit from `dict` (or `UserDict`) and overwrite a bunch of the methods
  - Raise an exception in case of methods that don't make sense
- Keep no `dict` functionality
  - I.e remove `FakeDict`
  - Force usage of the fields/attributes
  - Where parts of the config required as `dict`, call `asdict`
- Keep partial backwards compatibility but encourage use of the new system
  - Perhaps overwrite
  - I.e warn when the dict-based access is attempted